### PR TITLE
Refactor LocationDetail to classbased view

### PR DESCRIPTION
**Pull request by jonathan-s**
This refactors LoctionDetail to be a class based view instead of function based view. It also introduces a new package `rules`. You can read more about it here: https://github.com/dfunckt/django-rules

Making any permission related logic becomes really clean. 

For this PR I also want to write tests. So I'm creating a draft pull request for now.

### CODE REVIEW

**Comment by jonathan-s** [comment]
Ready for review now @bfirsh

File: modernomad/core/templates/location/location_detail.html [hunk 1]
```diff
@@ -108,7 +108,7 @@ <h3><a href="stay" class="btn btn-lg btn-primary btn-brand">View all rooms</a></
                     if(this.id == 'arrive_date') {
                         startDate = $('#arrive_date').datepicker("getDate");
                           var rMin = new Date(startDate.getFullYear(), startDate.getMonth(),startDate.getDate() + 1); // Min Date = Selected + 1d
-                          var rMax = new Date(startDate.getFullYear(), startDate.getMonth(),startDate.getDate() + {{max_days}} ); // Max Date = Selected + max_days
+                          var rMax = new Date(startDate.getFullYear(), startDate.getMonth(),startDate.getDate() + {{location.max_days}} ); // Max Date = Selected + max_days
```
Comment (bfirsh) [inline modernomad/core/templates/location/location_detail.html]: I think tests are failing due to this variable being unset.

File: modernomad/settings/local.py [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
+from .common import INSTALLED_APPS, MIDDLEWARE_CLASSES
 
 DEBUG = True
-INSTALLED_APPS = INSTALLED_APPS + [
-    'debug_toolbar'
-]
+TESTING_MODE = 'test' in sys.argv
+DEV_MODE = DEBUG and not TESTING_MODE
+
+if DEV_MODE:
```
Comment (bfirsh) [inline modernomad/settings/local.py]: These environment variables took a bit to get my head round. Perhaps we can avoid the extra `DEV_MODE` variable by just having `if not TESTING_MODE` here? That perhaps better explains the intention when reading the code -- don't load the debug toolbar when running tests.

File: modernomad/settings/local.py [hunk 3]
```diff
@@ -1,16 +1,21 @@
+import sys
 from .common import *  # noqa
-from .common import INSTALLED_APPS
+from .common import INSTALLED_APPS, MIDDLEWARE_CLASSES
 
 DEBUG = True
-INSTALLED_APPS = INSTALLED_APPS + [
-    'debug_toolbar'
-]
+TESTING_MODE = 'test' in sys.argv
```
Comment (bfirsh) [inline modernomad/settings/local.py]: A bikeshed: `RUNNING_TESTS`? Then the statement below becomes `if not RUNNING_TESTS` which reads quite nice.

**Comment by jonathan-s** [comment]
@bfirsh All passing now.

File: modernomad/urls/main.py [hunk 4]
```diff
@@ -48,7 +48,7 @@
     url(r'^%s/(?P<path>.*)$' % media_url, django.views.static.serve, {'document_root': settings.MEDIA_ROOT, 'show_indexes': True}),
 ]
 
-if settings.DEBUG:
+if settings.DEV_MODE:
```
Comment (bfirsh) [inline modernomad/urls/main.py]: This isn't set now, I think...

Comment (jonathan-s) [reply modernomad/urls/main.py]: @bfirsh Thanks, I had missed that one.

**Comment by bfirsh** [comment]
🙌🎉

**Pull request merged**